code/api/views.py

```diff
@@ -46,13 +47,20 @@ class AuthorsView(View):
 
     @method_decorator(ensure_csrf_cookie)
     def get(self, request):
-        """ GET - Retrieve all user profiles """
-
+        """ GET - Retrieve all user profiles 
+            'page' is indexed from 1, NOT 0.
+            'size' must be greater than 0
+        """
+        authors = LocalAuthor.objects.order_by('pk')
         page = request.GET.get("page")
         size = request.GET.get("size")
+        
+        if page and size:
+            page = int(page)
+            size = int(size)
+            authors = getPaginated(authors, page, size)
 
-        authors = [author.as_json() for author in LocalAuthor.objects.all()]
-
+        authors = [author.as_json() for author in authors]
```

Comment:
`getPaginated` returns an `HttpResponseBadRequest` object if page < 1 or size < 1, but then this get's assigned to `author`. Then the code crashes on `author.as_json()`

Reply:
Moved error handling to views.py